core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/EncodingStrategy.java

```diff
@@ -112,10 +112,41 @@ public ColumnPageDecoder createDecoderLegacy(ValueEncoderMeta metadata) {
       case SHORT:
       case INT:
       case LONG:
-        return DefaultEncodingStrategy.selectCodecByAlgorithmForIntegral(stats).createDecoder(null);
+        // create the codec based on algorithm and create decoder by recovering the metadata
+        ColumnPageCodec codec = DefaultEncodingStrategy.selectCodecByAlgorithmForIntegral(stats);
+        if (codec instanceof AdaptiveIntegralCodec) {
+          AdaptiveIntegralCodec adaptiveCodec = (AdaptiveIntegralCodec) codec;
+          AdaptiveIntegralEncoderMeta meta = new AdaptiveIntegralEncoderMeta(
```

Comment:
The parameter sequence of `DataType`, `stats`,`compressorName` are different from that in `AdaptiveDeltaIntegralEncoderMeta` and `DirectCompressorEncoderMeta`. It would be better to keep them same.

Reply:
It is  because AdaptiveIntegralEncoderMeta constructor is different from other. I will modify this